lib/v2/screens/sign_up/create_username/create_username.dart

```diff
@@ -82,18 +99,40 @@ class _CreateUsernameState extends State<CreateUsername> {
     );
   }
 
+  String? _validateUsername(String? username) {
+    final validCharacters = RegExp(r'^[a-z1-5]+$');
+
+    if (username.isNullOrEmpty) {
+      return 'Please select a username';
+    } else if (RegExp(r'0|6|7|8|9').allMatches(username!).isNotEmpty) {
+      return 'Only numbers 1-5 can be used.';
+    } else if (username.toLowerCase() != username) {
+      return "Name can be lowercase only";
+    } else if (!validCharacters.hasMatch(username) || username.contains(' ')) {
+      return "No special characters or spaces can be used";
+    } else if (username.length != 12) {
+      return 'Username must be 12 characters long';
+    }
+
+    return null;
+  }
+
   void _onUsernameChanged(String text) {
-    _debouncer.run(() {
-      _bloc.add(OnUsernameChanged(userName: text));
+    // This is to update the next button when we go from valid username to invalid
+    setState(() {
+      if (_usernameFormKey.currentState!.validate()) {
+        _bloc.add(OnUsernameChanged(username: text));
+      }
```

Comment:
You can wrap the `FlatButtonLong` in a `Blocbuilder` to rebuild the button enable/disable, also for the sake of keeping the use of the bloc uniform in the proejct.

Let's try to use setState only for very necessary cases.

Reply:
It's already wrapped in `BlocConsumer` but I moved the validation into bloc which resolved this.